data-protocols/dsp/dsp-negotiation/dsp-negotiation-api/src/main/java/org/eclipse/edc/protocol/dsp/negotiation/api/controller/DspNegotiationController.java

```diff
@@ -0,0 +1,282 @@
+import static org.eclipse.edc.protocol.dsp.negotiation.spi.DspNegotiationPropertyAndTypeNames.DSPACE_PREFIX;
+import static org.eclipse.edc.protocol.dsp.negotiation.spi.DspNegotiationPropertyAndTypeNames.DSPACE_SCHEMA;
+import static org.eclipse.edc.protocol.dsp.negotiation.spi.NegotiationApiPaths.AGREEMENT;
+import static org.eclipse.edc.protocol.dsp.negotiation.spi.NegotiationApiPaths.BASE_PATH;
+import static org.eclipse.edc.protocol.dsp.negotiation.spi.NegotiationApiPaths.CONTRACT_OFFER;
+import static org.eclipse.edc.protocol.dsp.negotiation.spi.NegotiationApiPaths.CONTRACT_REQUEST;
+import static org.eclipse.edc.protocol.dsp.negotiation.spi.NegotiationApiPaths.EVENT;
+import static org.eclipse.edc.protocol.dsp.negotiation.spi.NegotiationApiPaths.INITIAL_CONTRACT_REQUEST;
+import static org.eclipse.edc.protocol.dsp.negotiation.spi.NegotiationApiPaths.TERMINATION;
+import static org.eclipse.edc.protocol.dsp.negotiation.spi.NegotiationApiPaths.VERIFICATION;
+import static org.eclipse.edc.protocol.dsp.transform.transformer.Namespaces.ODRL_PREFIX;
+import static org.eclipse.edc.protocol.dsp.transform.transformer.Namespaces.ODRL_SCHEMA;
+import static org.eclipse.edc.protocol.dsp.transform.util.TypeUtil.isOfExpectedType;
+import static org.eclipse.edc.web.spi.exception.ServiceResultHandler.exceptionMapper;
+
+/**
+ * Provides consumer and provider endpoints according to http binding of the dataspace protocol.
+ */
+@Consumes({MediaType.APPLICATION_JSON})
+@Produces({MediaType.APPLICATION_JSON})
+@Path(BASE_PATH)
+public class DspNegotiationController implements DspNegotiationApi {
+
+    private final IdentityService identityService;
+    private final JsonLdTransformerRegistry transformerRegistry;
+    private final String callbackAddress;
+    private final Monitor monitor;
+    private final ContractNegotiationService service;
+    private final ContractNegotiationProtocolService protocolService;
+    private final ObjectMapper mapper;
+
+    public DspNegotiationController(Monitor monitor, TypeManager typeManager, String callbackAddress,
+                                    IdentityService identityService, JsonLdTransformerRegistry transformerRegistry,
+                                    ContractNegotiationService service, ContractNegotiationProtocolService protocolService) {
+        this.callbackAddress = callbackAddress;
+        this.identityService = identityService;
+        this.monitor = monitor;
+        this.protocolService = protocolService;
+        this.service = service;
+        this.transformerRegistry = transformerRegistry;
+
+        this.mapper = typeManager.getMapper(TYPE_MANAGER_CONTEXT_JSON_LD);
+    }
+
+    @GET
+    @Path("{id}")
+    @Override
+    public Map<String, Object> getNegotiation(@PathParam("id") String id, @HeaderParam(HttpHeaders.AUTHORIZATION) String token) {
+        monitor.debug(format("DSP: Incoming request for contract negotiation with id %s", id));
+
```

Comment:
This method needs to check that the claims match the stored counter party on the negotiation. Otherwise, someone can view a CN that they are not a party tyo if they know the id.

Reply:
Created #2795 and added `UnsupportedOperationException` to the endpoint. Same as for the transfer process.